modules/actions/pool/sync-join-exits-v2.ts

```diff
@@ -1,114 +1,92 @@
-//             },
-//             chain: chain,
-//             vaultVersion,
-//         },
-//     });
+    // Store only the events that are not already in the DB
+    const existingEvents = await prisma.poolEvent.findMany({
+        where: {
+            id: { in: joinExits.map((event) => event.id) },
+            type: {
+                in: ['JOIN', 'EXIT'],
+            },
+            chain: chain,
+            vaultVersion,
+        },
+    });
 
-//     const events = joinExits.filter((event) => !existingEvents.some((existing) => existing.id === event.id));
+    const events = joinExits.filter((event) => !existingEvents.some((existing) => existing.id === event.id));
 
-//     // Prepare DB entries
-//     const dbEntries = await Promise.all(
-//         events.map(async (event) => {
-//             // TODO: Calculate USD amounts with token prices at the time of the event
-//             // 🚨 Reading from the DB in a loop – will get slow with a large events volume
-//             // But we need prices based on the event timestamp, so batching this should be based on timestamp ranges
-//             const prices = await prisma.prismaTokenPrice.findMany({
-//                 where: {
-//                     tokenAddress: { in: event.pool.tokensList },
-//                     timestamp: roundToHour(Number(event.timestamp)), // 🚨 Assuming all prices are available hourly
-//                     chain: chain,
-//                 },
-//                 include: {
-//                     token: true,
-//                 },
-//             });
+    // Prepare DB entries
+    const dbEntries = events.map((event) => ({
+        vaultVersion,
+        id: event.id, // tx + logIndex
+        tx: event.tx,
+        type: event.type === 'Join' ? PoolEventType.JOIN : PoolEventType.EXIT,
+        poolId: event.pool.id,
+        chain: chain,
+        userAddress: event.sender,
+        blockNumber: 0, // TODO: fix fantom subgraph to include blocknumber // Number(event.block),
+        blockTimestamp: Number(event.timestamp),
+        logIndex: Number(event.id.substring(66)),
+        valueUSD: 0,
+        payload: {
+            tokens: event.pool.tokensList.map((token, i) => ({
+                address: token,
+                amount: event.amounts[i],
+                valueUSD: 0,
+            })),
+        },
+    }));
 
-//             const usd = event.pool.tokensList.map((address, index) => {
-//                 const price = prices.find((price) => price.tokenAddress === address);
-//                 return {
-//                     address: address,
-//                     amount: event.amounts[index],
-//                     valueUSD: Number(event.amounts[index]) * (price?.price || 0), // TODO: check USD amount
-//                 };
-//             });
+    // Enrich with USD values
+    // TODO: do we need a separate function to update prices? If so, we should be syncing events first, then running a price on them
```

Comment:
I dont think we do, it's fine like this. We dont need to update prices later, as we want prices at event time.

Reply:
The main reason for separating enrichment is some errors along the way, or a need for redoing this step, for example when some prices where messed up and need to be recalculated after fixing them.

Reply:
hm... leave the todo, but wouldn't add that func yet